### ISSUE TITLE
login: Give useful hints, format help, and error feedback in UI

### ISSUE THREAD

**Issue by gnprice** [zulip/zulip-flutter#35]
This will be a followup to #11, where I think for the first iteration we'll leave this aspect out, or at least incomplete.

When the user is trying to log into an account, we should:
* accept a URL with the leading `https://` left implicit;
* give useful hints about what the "Zulip server URL" field is asking for;
* give useful feedback in the UI when the URL is ill-formed;
* give useful feedback when the URL isn't of a Zulip server, or the server can't be reached;
* give useful feedback when the email they try to enter isn't in the shape of an email;
* give useful feedback when the server doesn't accept the login;
* perhaps more in the same vein.

In particular we should bring this to at least the same level of helpfulness as we currently have in the RN app.

**Comment by chrisbobbe** [comment]
With #87 and #88 merged, here are the remaining things I think we might want to do (which could be tracked here or in new issue(s)):

- [ ] #91
- [ ] (After #91) When `requireEmailFormatUsernames` is true, make form-validation check that `username` is shaped like an email address
- [ ] This TODO on our `getServerSettings` request:
  ```dart
  // TODO(#35): Perhaps detect realmless root domain, for more specific onboarding feedback.
  //   See thread, and the zulip-mobile code and chat thread it links to:
  //     https://github.com/zulip/zulip-flutter/pull/55#discussion_r1160267577
  ```
- [ ] This TODO in error handling on our `getServerSettings` request:
  ```dart
          // TODO(#35) give more helpful feedback; see `fetchServerSettings`
          //   in zulip-mobile's src/message/fetchActions.js. Needs #37.
          showErrorDialog(context: context,
            title: 'Could not connect', message: 'Failed to connect to server:\n$url');
  ```
- [ ] This TODO in our error handling on `fetchApiKey`:
  ```dart
          // TODO(#35) give more helpful feedback. Needs #37. The RN app is
          //   unhelpful here; we should at least recognize invalid auth errors, and
          //   errors for deactivated user or realm (see zulip-mobile#4571).
          showErrorDialog(context: context, title: 'Login failed');
  ```
- [ ] This TODO on inserting an account into the local database:
  ```dart
  // TODO(#35): give feedback to user on SQL exception, like dupe realm+user
  ```
- [ ] Help users learn what a "server URL" is, with a link to the appropriate doc, like in zulip/zulip-mobile@52288facc
- [ ] If it's been helpful in zulip-mobile (I've used it), give clickable "suggestions" based on text typed so far, if it seems like the user might want CZO or a Zulip Cloud realm.
- [ ] For username/password autofill, find out if we can tell the platform to highlight Zulip Cloud and CZO credentials, perhaps after demonstrating that we, the app publishers, also control those web domains.
